src/facet.js

```diff
@@ -155,6 +147,10 @@ function and(a, b) {
   };
 }
 
+function isEmptyFacet(facets, {x, y}) {
```

Comment:
This code de-deplication isn’t related to this PR; it would be easier to review if you just made the minimal change necessary to fix the bug (adding the early return if (empty)).